packages/examples/packages/cronjobs/src/index.ts

```diff
@@ -29,6 +32,57 @@ export const onCronjob: OnCronjobHandler = async ({ request }) => {
           ]),
         },
       });
+    case 'fireNotification':
+      return snap.request({
+        method: 'snap_notify',
+        params: {
+          type: 'inApp',
+          message: 'Hello world!',
+        },
+      });
+    default:
+      // eslint-disable-next-line @typescript-eslint/no-throw-literal
+      throw new MethodNotFoundError({ method: request.method });
+  }
+};
+
+/**
+ * Handle incoming JSON-RPC requests from the dapp, sent through the
+ * `wallet_invokeSnap` method. This handler handles three methods:
+ *
+ * - `scheduleNotification`: Schedule a notification in the future.
+ * - `cancelNotification`: Cancel a notification.
+ * - `getBackgroundEvents`: Get the Snap's background events.
+ *
+ * @param params - The request parameters.
+ * @param params.request - The JSON-RPC request object.
+ * @returns The JSON-RPC response.
+ * @see https://docs.metamask.io/snaps/reference/exports/#onrpcrequest
+ * @see https://docs.metamask.io/snaps/reference/rpc-api/#wallet_invokesnap
+ */
+export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
+  switch (request.method) {
+    case 'scheduleNotification':
+      return snap.request({
+        method: 'snap_scheduleBackgroundEvent',
+        params: {
+          date: (request.params as Record<string, string>).date,
```

Comment:
Can you follow the same pattern as other examples instead of casting here?

Reply:
Other examples are casting as well, I defined the types though.